src/WebApi/Controllers/StateController.cs

```diff
@@ -4,79 +4,129 @@
 using Microsoft.AspNetCore.Authorization;
 using Microsoft.AspNetCore.Mvc;
 
-namespace Devpro.TerraformBackend.WebApi.Controllers
+namespace Devpro.TerraformBackend.WebApi.Controllers;
+
+[Authorize]
+[ApiController]
+[Route("{tenant}/state")]
+public class StateController(IStateRepository stateRepository, IStateLockRepository stateLockRepository) : ControllerBase
 {
-    [Authorize]
-    [ApiController]
-    [Route("state")]
-    public class StateController : ControllerBase
+    /// <summary>
+    /// Get Terraform state value.
+    /// GET /:tenant/state/:name?ID=:lockId
+    /// </summary>
+    /// <param name="tenant"></param>
+    /// <param name="name">The name of the Terraform state</param>
+    /// <returns>Raw string</returns>
+    [HttpGet("{name:regex([[a-zA-Z]]+)}", Name = "GetState")]
```

Comment:
@devpro Why only a-z (and their uppercase variants)? Wouldn't it be better to use a typed value which will be checked within the type. This will make it testable and most likely more performant as there is no guarantee how many times this regex is created when in the case of typed class it needs to be instantiated only once during the lifetime.